prometheus-to-sd/translator/metric_descriptor_cache.go

```diff
@@ -66,6 +66,27 @@ func (cache *MetricDescriptorCache) MarkStale() {
 	cache.fresh = false
 }
 
+// ValidateMetricDescriptors checks if metric descriptors differs from the values kept in the cache.
+// If the value has changed then metric family is marked is broken. Use this method to verify that
+// metrics with prefix "container.googleapis.com" haven't changed.
+func (cache *MetricDescriptorCache) ValidateMetricDescriptors(metrics map[string]*dto.MetricFamily, whitelisted []string) {
+	// Perform cache operation only if cache was recently refreshed. This is done mostly from the optimization point
+	// of view, we don't want to check all metric descriptors too often, as they should change rarely.
+	if !cache.fresh {
+		return
+	}
+	for _, metricFamily := range metrics {
+		if isMetricWhitelisted(metricFamily.GetName(), whitelisted) {
+			metricDescriptor, ok := cache.descriptors[metricFamily.GetName()]
+			updatedMetricDescriptor := MetricFamilyToMetricDescriptor(cache.config, metricFamily, metricDescriptor)
+			if ok && descriptorChanged(metricDescriptor, updatedMetricDescriptor) {
+				cache.broken[metricFamily.GetName()] = true
+				glog.Warningf("Definition of the metric %s was changed and metric is not going to be pushed", metricFamily.GetName())
```

Comment:
How often is this going to be logged?

Reply:
I see your point. It's going to be logged every 10 minutes (configured by metric-descriptors-resolution flag) for every broken metric family. If you think that it's too much then I can decrease log level for it.